### ISSUE TITLE
Memory space interface design

### ISSUE THREAD

**Issue by anagainaru** [anagainaru/ADIOS2-addons#61]
**Interface for setting the memory space**

Change from
```c++
bpWriter.Put(data, gpuSimData, adios2::Mode::Deferred,
                       adios2::MemorySpace::CUDA);
```
to
```c++
data.SetMemorySpace(adios2::MemorySpace::CUDA);
bpWriter.Put(data, gpuSimData);
```

**Comment by anagainaru** [comment]
```c++
variable.SetMemorySpace(mem);
```

stores a bool (true or false) based on the memory space, should store the actual memory space and IsGPU will check for host

**Comment by anagainaru** [comment]
Closed by Issue #62
